### PR TITLE
Added navigation drawer and top app bar

### PR DESCRIPTION
**Pull request by ahmedm-git**

### CODE REVIEW

File: app/src/main/java/com/example/androidapp/MainActivity.kt [hunk 1]
```diff
@@ -1,36 +1,51 @@
...
-        appBarConfiguration = AppBarConfiguration(navController.graph)
-        setupActionBarWithNavController(navController, appBarConfiguration)
+        // Load the recipes page by default.
+        supportFragmentManager.beginTransaction().replace(R.id.fragment_container, FragmentRecipes()).commit()
 
-        binding.fab.setOnClickListener { view ->
-            Snackbar.make(view, "Replace with your own action", Snackbar.LENGTH_LONG)
-                .setAction("Action", null).show()
+        // Set up listeners for the bottom navigation drawer.
+        navbarView.setOnNavigationItemSelectedListener { item ->
```
Comment (kertox662) [inline app/src/main/java/com/example/androidapp/MainActivity.kt:33]: Looks like this is deprecated, replaced by ` NavigationBarView.setOnItemSelectedListener(NavigationBarView.OnItemSelectedListener)` instead. Could you change that?

File: app/src/main/java/com/example/androidapp/MainActivity.kt [hunk 2]
```diff
@@ -1,36 +1,51 @@
...
 
-        setSupportActionBar(binding.toolbar)
+        navbarView = findViewById(R.id.bottom_navigation)
+        fragmentContainer = findViewById(R.id.fragment_container)
+        topAppBar = findViewById(R.id.top_app_bar)
 
-        val navController = findNavController(R.id.nav_host_fragment_content_main)
-        appBarConfiguration = AppBarConfiguration(navController.graph)
-        setupActionBarWithNavController(navController, appBarConfiguration)
+        // Load the recipes page by default.
```
Comment (kertox662) [inline app/src/main/java/com/example/androidapp/MainActivity.kt:29]: Thanks for the comments.

File: app/src/main/res/drawable/account_icon.xml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (kertox662) [inline app/src/main/res/drawable/account_icon.xml:1]: I'm curious, where did you get these? Looked good, hopefully from the android icons or copyright free source?

File: app/src/main/res/menu/bottom_navigation_menu.xml [hunk 4]
```diff
@@ -0,0 +1,28 @@
+<?xml version="1.0" encoding="utf-8"?>
+<menu xmlns:android="http://schemas.android.com/apk/res/android">
+  <item
+      android:id="@+id/page_1"
```
Comment (kertox662) [inline app/src/main/res/menu/bottom_navigation_menu.xml:4]: Didn't see where this is being used exactly. It looked like some list, but maybe its for the nav bar. Could you raname these to more exact names like `page_recipes`?

**Review by kertox662 — COMMENTED** [review]
Looks good overall and feels nice. Feel free to merge if everything's good.

File: app/src/main/res/drawable/account_icon.xml (hunk 3 above)
Comment (ahmedm-git) [reply app/src/main/res/drawable/account_icon.xml:1]: I got it from https://fonts.google.com/icons

File: app/src/main/res/menu/bottom_navigation_menu.xml (hunk 4 above)
Comment (ahmedm-git) [reply app/src/main/res/menu/bottom_navigation_menu.xml:4]: Yeah, this is where the items in the navigation bar are defined. I'll rename them to be more clear.

**Pull request merged**